### DISCUSSION TITLE
Is there an API call to directly map string values to form fields?

### DISCUSSION THREAD

**Discussion by kharabela** [pdfcpu/pdfcpu#699] [Q&A]
// Merges form fields in the source pdf and dumps results in the provided directory.
func MergeFormFields() {

	inDir := filepath.Join("pdffill")
	csvDir := filepath.Join("pdffill")
	outDir := filepath.Join("pdffill", "merge")

	for _, tt := range []struct {
		msg       string
		inFile    string
		inFileCSV string
	}{
		{"TestMultiFillFormCSVEnglish", "english.pdf", "english.csv"},
	} {

		inFile := filepath.Join(inDir, tt.inFile)
		inFileCSV := filepath.Join(csvDir, tt.inFileCSV)
		if err := pdfcpu.MultiFillFormFile(inFile, inFileCSV, outDir, inFile, true, pdfcpu.LoadConfiguration()); err != nil {
			fmt.Println(err)
		}
	}

**Comment by hhrutter** [comment]
Hello!
Please check out the content of *english.csv* for this test.
The header line identifies the field names for filling:
```
"firstName1","lastName1","dob1","gender1","note1","city11","city12","@img(page:1, pos:br, off:-30 30, w:100)"
"Jane","Doe","06.01.2000","female","Person #1","San Francisco","London","../../testdata/resources/qr.png,(https://pdfcpu.io)"
"Joe","Doe","30.07.2001","male","Person #2","São Paulo","San Francisco","../../testdata/resources/logoVerySmall.png,(https://pdfcpu.io)"
"Jackie","Doe","29.11.1965","non-binary","Person #3","Vienna","Sidney","../../testdata/resources/pdfchip3.png,(https://pdfcpu.io)"
```

**Reply by kharabela** [reply]
Thanks for the response. I got my merge field name & string value that I can map.What method/function should I use to directly map it.My use case is not to use csv file.

**Comment by hhrutter** [comment]
If you don't want to us either JSON or CSV (which I recommend also for form documentation purposes)
you can call `ExportForm` and then feed the resulting `FormGroup` into `FillForm`.

**Comment by hhrutter** [comment]
Closing this as answered.

For programmatic form filling without CSV/JSON, the intended API path is to export the form model first and then feed the populated model back into the filler:

ExportForm → populate/update the resulting FormGroup in Go → FillForm

CSV/JSON remain the recommended formats when you also want a stable external representation or form-filling documentation, but they are not required for purely programmatic use.
